ste/init.go

```diff
@@ -453,6 +453,9 @@ func GetJobSummary(jobID common.JobID) common.ListJobSummaryResponse {
 		js.PercentComplete = 100
 	} else {
 		js.PercentComplete = 100 * float32(js.TotalBytesTransferred) / float32(js.TotalBytesExpected)
+		if js.PercentComplete > 100 {
+			js.PercentComplete = 100 // we had some transient conditions that temporarily gave values above 100
```

Comment:
Uhhh... This seems like a really hacky solution. What were the underlying conditions that would cause values above 100%? I don't feel like flashing 100% for a few seconds and then going back down (if that's even possible) once those conditions clear solves the problem.

Reply:
I don't know what the underlying conditions are. My repro that was working a few weeks ago didn't work when I tried it again. You're right that its hacky and ugly, for all the reasons you said.  However, I don't feel I have time to do any better in the current release cycle.

Reply:
Oh, I've discovered another way to reproduce it, and a more persistent (and therefore debuggable) manner.  I will investigate.

Reply:
That's fair. If it's not feasible to get done right now, I'm fine with a hacky solution on something this non-critical for a little while.

Reply:
It's indeed a bit hacky, but I'm ok with following up with a permanent solution in the next release.

Reply:
I've found and fixed it. There were two problems. The main one was that I was deliberately not taking into account the fact that the last chunk in each file may be smaller than the block size.  I thought it wouldn't make much difference - but as we've seen in recent days, there are cases where it can.  The second was some badly chosen integer types - surprisingly its not an error in Go to negate an unsigned int, and pass the result to another unsigned int.

Reply:
Good job!

I'm actually not surprised by the lack of erroring on that in Go. It doesn't error out on integer over-(or under)-flows.